frontend/components/Documents.js

```diff
@@ -56,6 +57,24 @@ const Documents = () => {
 	    }));
     };
 
+    const handleAttributeInputChange = (event, index) => {
+        const {name, value} = event.target;
+        console.log(name, value);
+        const attributesList = [...newAttributes];
+        attributesList[index][name] = value;
+        setNewAttributes(attributesList);
```

Comment:
I think line 63 isn't quite doing what it should in this case. The spread syntax doesn't deep copy, so the copied objects inside that array are references. That means that line 64 actually updates state without using the `set` function. Something like this might do the trick:

```
setNewAttributes(previousAttributes => {
  return previousAttributes.map((attribute, attribute_index) => {
    return attribute_index === index ? {name, value}: attribute;
  };
});
```

Reply:
updated!